### PR TITLE
DHFPROD-5022: Docs: Hub Central - Update tooltips.config.js

### PR DESCRIPTION
**Pull request by devinnicholson**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub-central/ui/src/config/tooltips.config.tsx [hunk 1]
```diff
@@ -0,0 +1,681 @@
...
+  targetDatabase: 'Choose where to store the processed data.',
+  targetFormat: 'Choose the format for the resulting documents.',
+  targetPermissions: 'Enter the permissions.',
+  headers: 'Enter additional header metadata.',
+  additionalCollections: 'Enter additional collection tags.',
+  provGranularity: 'Choose the level of provenance detail.',
+  validateEntity: 'Choose how to handle documents with errors.',
+  batchSize : 'Enter the batch size.',
+  processors: 'Enter the array of processor settings.',
+    /* Can we use an example like this?
```
Comment (xnikhil08) [inline marklogic-data-hub-central/ui/src/config/tooltips.config.tsx:140]: Let's not keep the code that is commented out and we don't need for now. Looks like this is a question for PM & UX team to include a sample processor and Custom hook in the tooltip and once its approved then we can add it later.

File: marklogic-data-hub-central/ui/src/config/tooltips.config.tsx [hunk 2]
```diff
@@ -0,0 +1,681 @@
...
+    '    "path": "/uri/of/custom/module/in/modules/database/a.sjs", \n' +
+    '    "vars": { \n' +
+    '      "myParameter": "myParameterValue" \n' +
+    '    } \n' +
+    '    "when": "beforeContentPersisted", \n' +
+    '  } \n' +
+    ']',
+    */
+  customHook: 'Enter the custom hook settings.',
+    /* Can we use an example like this?
```
Comment (xnikhil08) [inline marklogic-data-hub-central/ui/src/config/tooltips.config.tsx:152]: Same comment as given earlier.

**Review by xnikhil08 — CHANGES_REQUESTED** [review]
Changes look good. 
I just see some code at multiple places in the form of comments and thats probably a question which should be clarified in the associated JIRA tickdet and once approved then let's add the help text later in the tooltips file.

**Comment by xnikhil08** [comment]
@devinnicholson - Looks like few unit and e2e tests are failing in the pipeline. Please fix those.

File: marklogic-data-hub-central/ui/src/config/messages.config.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+const AdvancedSettingsMessages = {
+    'targetPermissions': {
+        'incorrectFormat' : (
+        <span>The format of the string is incorrect. The required format is 
+            <span style={{fontFamily: "monospace"}}> role, capability, role, capability,....</span> 
+        </span>),
+        'invalidCapabilities' : (
+        <span>The string contains invalid capabilities. Capabilities must be 
+            <span style={{fontFamily: "monospace"}}> read, insert, update, or execute</span>.
```
Comment (wooldridge) [inline marklogic-data-hub-central/ui/src/config/messages.config.tsx:11]: @devinnicholson, the invalidCapabilities monospaced text needs to be updated slightly based on what Eleanor provided, it should look like the following:

<span style={{fontFamily: "monospace"}}>read</span>, <span style={{fontFamily: "monospace"}}>insert</span>, <span style={{fontFamily: "monospace"}}>update</span>, or <span style={{fontFamily: "monospace"}}>execute</span>.

**Review by wooldridge — CHANGES_REQUESTED** [review]

**Comment by wooldridge** [comment]
@devinnicholson, I'm seeing the unit tests fail that inspect the target perms text. 
```
Summary of all failing tests
 FAIL  src/components/load/load-list.test.tsx (37.805s)
  ● Load data component › Verify Load settings from list view renders correctly

    expect(element).toHaveTextContent()

    Expected element to have text content:
      <span>The format of the string is incorrect. The required format is<span style={{"fontFamily": "monospace"}}> role, capability, role, capability,....</span></span>
    Received:
      The format of the string is incorrect. The required format is role, capability, role, capability,....

      180 |     expect(targetPermissions).toHaveValue("role1");
      181 |     fireEvent.blur(targetPermissions);
    > 182 |     expect(getByTestId("validationError")).toHaveTextContent(AdvancedSettingsMessages.targetPermissions.incorrectFormat);
          |                                            ^
      183 | 
      184 |     fireEvent.change(targetPermissions, {target: {value: "role1,reader"}}); // BAD permissions
      185 |     expect(targetPermissions).toHaveValue("role1,reader");

      at Object.<anonymous> (src/components/load/load-list.test.tsx:182:44)

 FAIL  src/components/steps/steps.test.tsx (6.29s)
  ● Steps settings component › Verify rendering of Mapping step, tab disabling on form error, discard changes dialog on close

    expect(element).toHaveTextContent()

    Expected element to have text content:
      <span>The format of the string is incorrect. The required format is<span style={{"fontFamily": "monospace"}}> role, capability, role, capability,....</span></span>
    Received:
      The format of the string is incorrect. The required format is role, capability, role, capability,....

      132 |     expect(getByPlaceholderText("Please enter target permissions")).toHaveValue("bad-value");
      133 |     fireEvent.blur(getByPlaceholderText("Please enter target permissions"));
    > 134 |     expect(getByTestId("validationError")).toHaveTextContent(AdvancedSettingsMessages.targetPermissions.incorrectFormat);
          |                                            ^
      135 | 
      136 |     expect(getByText("Basic").closest("div")).toHaveClass("ant-tabs-tab-disabled");
      137 |     fireEvent.mouseOver(getByText("Basic"));

      at Object.<anonymous> (src/components/steps/steps.test.tsx:134:44)

 FAIL  src/components/advanced-settings/advanced-settings.test.tsx (16.352s)
  ● Advanced step settings › Verify form fields can be input/selected

    expect(element).toHaveTextContent()

    Expected element to have text content:
      <span>The format of the string is incorrect. The required format is<span style={{"fontFamily": "monospace"}}> role, capability, role, capability,....</span></span>
    Received:
      The format of the string is incorrect. The required format is role, capability, role, capability,....

      294 |     expect(getByPlaceholderText("Please enter target permissions")).toHaveValue("data-hub-operator");
      295 |     fireEvent.blur(getByPlaceholderText("Please enter target permissions"));
    > 296 |     expect(getByTestId("validationError")).toHaveTextContent(AdvancedSettingsMessages.targetPermissions.incorrectFormat);
          |                                            ^
      297 | 
      298 |     fireEvent.change(getByPlaceholderText("Please enter target permissions"), {target: {value: "data-hub-operator,read"}});
      299 |     expect(getByPlaceholderText("Please enter target permissions")).toHaveValue("data-hub-operator,read");

      at Object.<anonymous> (src/components/advanced-settings/advanced-settings.test.tsx:296:44)


Test Suites: 3 failed, 76 passed, 79 total
Tests:       3 failed, 2 skipped, 394 passed, 399 total
Snapshots:   0 total
Time:        101.888s
```
Are you seeing this? You probably have to add the string value to check directly in the test (not via the file reference) since it appears the test framework does not recognize the added markup.

**Comment by xnikhil08** [comment]
@devinnicholson - Few e2e/unit tests are still failing in the pipeline. Please look into those.

**Comment by wooldridge** [comment]
@devinnicholson, here's an e2e step that is failing (see: https://jenkins.marklogic.com/blue/organizations/jenkins/Datahub_CI/detail/PR-5051/15/tests):
```
Error
Timed out retrying: Expected to find content: 'The Match step defines criteria for determining whether the values from entiies match, and the action to take based on how close of a match they are.' but never did.
Stacktrace
AssertionError: Timed out retrying: Expected to find content: 'The Match step defines criteria for determining whether the values from entiies match, and the action to take based on how close of a match they are.' but never did.
    at Context.eval (http://localhost:8080/__cypress/tests?p=cypress/integration/curation/curate/matching.spec.tsx:149:12)
```
I think this is the relevant e2e code:

https://github.com/marklogic/marklogic-data-hub/blob/develop/marklogic-data-hub-central/ui/e2e/cypress/integration/curation/curate/matching.spec.tsx#L69

It needs to match what is displayed here, which is a reference to the new tooltip content:

https://github.com/marklogic/marklogic-data-hub/blob/develop/marklogic-data-hub-central/ui/src/components/entities/matching/matching-step-detail/matching-step-detail.tsx#L240

Optimally, the tile view and the test should be referencing the same content, instead of hardcoding.

**Review by wooldridge — CHANGES_REQUESTED** [review]
See my e2e comment.

**Review by wooldridge — CHANGES_REQUESTED** [review]
@devinnicholson, looks like there are linting issues in this branch:
```
/Users/mwooldri/projects/marklogic-data-hub/marklogic-data-hub-central/ui/src/components/advanced-settings/advanced-settings.test.tsx
  8:9  error  'AdvancedSettingsMessages' is defined but never used  @typescript-eslint/no-unused-vars

/Users/mwooldri/projects/marklogic-data-hub/marklogic-data-hub-central/ui/src/components/entities/mapping/mapping-card.test.tsx
  139:33  error  'debug' is assigned a value but never used  @typescript-eslint/no-unused-vars

/Users/mwooldri/projects/marklogic-data-hub/marklogic-data-hub-central/ui/src/components/load/load-list.test.tsx
    8:9  error  'AdvancedSettingsMessages' is defined but never used  @typescript-eslint/no-unused-vars
  181:1  error  Trailing spaces not allowed                           no-trailing-spaces
  188:1  error  Trailing spaces not allowed                           no-trailing-spaces

/Users/mwooldri/projects/marklogic-data-hub/marklogic-data-hub-central/ui/src/components/steps/steps.test.tsx
  7:9  error  'AdvancedSettingsMessages' is defined but never used  @typescript-eslint/no-unused-vars

✖ 6 problems (6 errors, 0 warnings)
  2 errors and 0 warnings potentially fixable with the `--fix` option.
```
Can you fix those issues, rebase, and force-push?

**Review by Sanjeevani19 — APPROVED** [review]
Looks good and tests passed on the pipeline.

**Review by briantang — APPROVED** [review]

**Review by xnikhil08 — APPROVED** [review]

**Review by wooldridge — APPROVED** [review]
@devinnicholson, it looks like CI/CD failed with some linting errors:
```
/Users/mwooldri/projects/marklogic-data-hub/marklogic-data-hub-central/ui/src/components/advanced-settings/advanced-settings.test.tsx
  8:9  error  'AdvancedSettingsMessages' is defined but never used  @typescript-eslint/no-unused-vars

/Users/mwooldri/projects/marklogic-data-hub/marklogic-data-hub-central/ui/src/components/load/load-list.test.tsx
    8:9  error  'AdvancedSettingsMessages' is defined but never used  @typescript-eslint/no-unused-vars
  181:1  error  Trailing spaces not allowed                           no-trailing-spaces
  188:1  error  Trailing spaces not allowed                           no-trailing-spaces

/Users/mwooldri/projects/marklogic-data-hub/marklogic-data-hub-central/ui/src/components/steps/steps.test.tsx
  7:9  error  'AdvancedSettingsMessages' is defined but never used  @typescript-eslint/no-unused-vars

✖ 5 problems (5 errors, 0 warnings)
  2 errors and 0 warnings potentially fixable with the `--fix` option.
```
Can you run linting with

**Pull request merged**